### DISCUSSION TITLE
Include files (Suricata.yaml)

### DISCUSSION THREAD

**Discussion by xfaith** [Security-Onion-Solutions/securityonion#7937] [Unsupported Versions] [answered]
2 questions really here about Suricata/Variables.

Looking for the proper place to annote an include file on the suricata.yaml, and then also the location to store said file.  In 16.04 I kept the file in the same folder as the suricata.yaml file.  Using this to do a mass rename/adding of variables to suricata.    

Side note , I tried to follow the docs to edit the suricata.yaml files (running a standalone) and the global, and minion sls files did not have any suricata details, unsure if i was supposed to add both.

**Comment by dougburks** [comment, marked as answer]
> Looking for the proper place to annote an include file on the suricata.yaml, and then also the location to store said file. In 16.04 I kept the file in the same folder as the suricata.yaml file. Using this to do a mass rename/adding of variables to suricata.

You should be able to add variables in the pillar as described here:
https://docs.securityonion.net/en/2.3/suricata.html#configuration

> Side note , I tried to follow the docs to edit the suricata.yaml files (running a standalone) and the global, and minion sls files did not have any suricata details, unsure if i was supposed to add both.

Just add to one or the other, not both. If it's a standalone box, it doesn't really matter which you use. If it's a distributed deployment, you can add to global so that it applies to all sensors or just to the minion if you want to modify per-sensor.

**Reply by xfaith** [reply]
Doug would that be ok for like 500 variables to be added that way?    I think I would still like to use an include file as it keeps it separate from the main functions of the system.    Maybe just add an include statement in the .sls file?

**Comment by cm-ops** [comment]
@xfaith if you wanted to use an include file for the Suricata variables you could do something like the following:
In `/opt/so/saltstack/local/pillar` create an sls file, for example, `surivars.sls`
Add the suricata config pillar in the file, example from the RTDs:
```
suricata:
  config:
    vars:
      address-groups:
        EXTERNAL_NET: "!$HOME_NET"
```
In the minion sls or global add the following (whatever you called your sls file above):
```
include:
  - surivars
```
Then restart Suricata on the node. That will pull in the variable changes/additions.

**Reply by xfaith** [reply]
So I added the include file via minions 
suricata:
  config:
    include:  file.yaml

But its looking for file.yaml @ /etc/suricata.    I am guessing this is in the docker as there is no /etc/suricata, but the docker container wont start either.    I had the file copied to /opt/so/conf/suricata (As this looks like the suricata.yaml that gets updated when you make the changes.

**Reply by xfaith** [reply]
A little more research with docker inspect so-suricata looks like the /opt/so/conf/suricata folder is the mount for most files (they are specifically called out here.  But I dont belive the link between say /opt/so/conf/suricata and /etc/suricata are linked.  

In case anyone wants to follow suit:   You can place your file in /opt/so/conf/suricata/rules/   (Other option was /nsm/suricata).  Now it might not be the best (I will wait for SecurityOnion team to chime in).  But you can add the follwing to your sls

Suricata:
  config:
    include: /etc/suricata/rules/file.yaml  ----  (/etc/suricata/rules/ (Docker) - is linked to the (OS) /opt/so/conf/suricata/rules
